### PR TITLE
Add flow types to components; Update prettier, eslint, and add babel integrations

### PR DESCRIPTION
**Pull request by ericliu121187**
https://cl.ly/1J1C3a0C1c1i <---- no more errors!

⚠️ Remember to `yarn`!

To run it the tests -- `yarn lint && yarn flow` should do the trick. Remember to do that before opening a PR.

* Add [flow](https://flow.org/en/) types
* Because we're using the latest version of Flow, after I added flow types, eslint was giving me an error that required us to install and bump up `babel-eslint`, `eslint-plugin-babel` -- https://github.com/eslint/eslint/issues/9767

### CODE REVIEW

File: frontend/.prettierrc [hunk 1]
```diff
@@ -1,6 +1,7 @@
 {
-  "printWidth": 80,
+  "arrowParens": "avoid",
   "parser": "flow",
+  "printWidth": 80,
```
Comment (ericliu121187) [inline frontend/.prettierrc]: just abc-ed this

File: frontend/src/components/App.js [hunk 2]
```diff
@@ -59,7 +64,7 @@ export default compose(
       const webSocket = new WebSocket(
         `ws://${window.location.hostname}:${PORT}/connect` // TODO move to constant
       );
-      webSocket.onmessage = function(msg) {
+      webSocket.onmessage = function() {
```
Comment (ericliu121187) [inline frontend/src/components/App.js]: remove unused var

File: frontend/src/components/LoginField.js [hunk 3]
```diff
@@ -28,7 +28,17 @@ const Username = styled.span`
...
 
-function LoginField({ handleChange, handleLoginSubmit, isLoggedIn, username }) {
+function LoginField({
+  handleChange,
+  handleLoginSubmit,
+  isLoggedIn,
+  username,
+}: {
+  handleChange: (SyntheticEvent<*>) => void,
+  handleLoginSubmit: (SyntheticEvent<*>) => void,
```
Comment (ericliu121187) [inline frontend/src/components/LoginField.js]: the parens were dictated by prettier

**Comment by txu89** [comment]
```
An unexpected error occurred: "https://repository.mark43.io/api/npm/npm-all/@babel/code-frame/-/code-frame-7.0.0-beta.40.tgz: Request failed \"401 Unauthorized\"".
```
This error appears when trying to update using yarn

**Comment by txu89** [comment]
Fixed yarn.lock using incorrect repository. Tested PR and no errors when `yarn lint` or `yarn flow`

**Comment by ericliu121187** [comment]
crap, thanks @txu89

**Pull request merged**